File: Python/Problem009.py
from math import floor, sqrt
from time import time

''' 
Problem 9: https://projecteuler.net/problem=9

A Pythagorean triplet is a set of three natural numbers, a < b < c, for which,
                    
                    a**2 + b**2 = c**2

There exists exactly one Pythagorean triplet for which a + b + c = 1000.
Find the product a*b*c.
'''

def isTriplet(a, b, c): return a < b < c and a**2 + b**2 == c**2 and a+b+c == 1000


def main():
    for i in range(1000):
        for j in range(1000):
            k = (i**2 + j**2) ** 0.5
            if isTriplet(i, j, int(k)):
                return i*j*int(k)


start = time()
ans = main()

if __name__ == '__main__':
    print(f"\nAnswer: { ans }")
    print(f"Time Taken: { time() - start }\n")

File: CPP/Problem009.cpp
#include "iostream"
#include "math.h"

int Problem009();
bool isTriplet(int a, int b, int c);

int main(){
    clock_t start; 
    start = std::clock();
    int ans = Problem009();
    double time_taken = double(clock() - start) / double(CLOCKS_PER_SEC); 

    printf("\nAnswer: %d\n", ans);
    printf("Time Taken: %f seconds\n\n", time_taken);

    return 0;
}

/*
Problem009 answers the problem at : https://projecteuler.net/problem=9

* Problem 9:
		There exists exactly one Pythagorean triplet for which a + b + c = 1000 (a < b < c).
		Find the product abc.
*/
int Problem009(){

    for (int i = 3; i < 500; i++){
        for (int j = i; j < 500; j++){
            int k = sqrt(i*i + j*j);
            if (isTriplet(i, j, k)){
                return i*j*k;
            }
        }
    }
    return 0;
}

bool isTriplet(int a, int b, int c){
    return (a<b && b<c) && (a*a + b*b == c*c) && (a+b+c == 1000);
}